tests/unit/multi_table/test_base.py

```diff
@@ -136,3 +142,223 @@ def test_get_metadata(self):
+            'upravna_enota': pd.DataFrame({
+                'id_upravna_enota': np.arange(10),
+            }),
+        }
+
+        instance = BaseMultiTableSynthesizer(metadata)
+
+        # Run and Assert
+        error_msg = "The provided data is missing the table {'nesreca'}."
+        with pytest.raises(InvalidDataError, match=error_msg):
+            instance.validate(data)
+
+    def test_validate_missing_multiple_tables(self):
+        """Test that an error is being raised when there are multiple tables missing."""
+        # Setup
+        metadata = get_multi_table_metadata()
+        data = {
+            'table1': pd.DataFrame({
+                'id_nesreca': np.arange(10),
+                'upravna_enota': np.arange(10),
+            }),
+            'table2': pd.DataFrame({
+                'upravna_enota': np.arange(10),
+                'id_nesreca': np.arange(10),
+            }),
+            'upravna_enota': pd.DataFrame({
+                'id_upravna_enota': np.arange(10),
+            }),
+        }
+
+        instance = BaseMultiTableSynthesizer(metadata)
+
+        # Run and Assert
+        with pytest.raises(InvalidDataError):
+            instance.validate(data)
+
+    def test_validate_data_is_not_dataframe(self):
+        """Test that an error is being raised when the data is not a dataframe."""
+        # Setup
+        metadata = get_multi_table_metadata()
+        data = {
+            'nesreca': pd.Series({
+                'id_nesreca': np.arange(10),
+                'upravna_enota': np.arange(10),
+            }),
+            'oseba': pd.DataFrame({
+                'upravna_enota': np.arange(10),
+                'id_nesreca': np.arange(10),
+            }),
+            'upravna_enota': pd.DataFrame({
+                'id_upravna_enota': np.arange(10),
+            }),
+        }
+
+        instance = BaseMultiTableSynthesizer(metadata)
+
+        # Run and Assert
+        error_msg = "Data must be a DataFrame, not a <class 'pandas.core.series.Series'>."
+        with pytest.raises(InvalidDataError, match=error_msg):
+            instance.validate(data)
+
+    def test_validate_data_does_not_match(self):
+        """Test that an error is being raised when the data does not match the metadata."""
+        # Setup
+        metadata = get_multi_table_metadata()
+        data = {
+            'nesreca': pd.DataFrame({
+                'id_nesreca': np.arange(10).astype(str),
+                'upravna_enota': np.arange(10),
+            }),
+            'oseba': pd.DataFrame({
+                'upravna_enota': np.arange(10).astype(str),
+                'id_nesreca': np.arange(10),
+            }),
+            'upravna_enota': pd.DataFrame({
+                'id_upravna_enota': np.arange(10).astype(str),
+            }),
+        }
+
+        instance = BaseMultiTableSynthesizer(metadata)
+
+        # Run and Assert
+        error_msg = re.escape(
+            'The provided data does not match the metadata:\n'
```

Comment:
This error message is a little weird. Maybe the colons at the end are supposed to be dots?

Reply:
This message comes from the `InvalidDataError` 🤔 I'm not sure if a new class is desired to be created with a different message. @amontanez24 any suggestions on that?

Reply:
I think it's helpful to have the label for each individual table since that helps the user know where to look. Maybe we can get rid of the top line. @npatki do you have thoughts on this?

Reply:
Is this for validating data against the metadata? The idea was to collect and print all the errors at once, which is why there was probably a colon?

```python
>>> synthesizer.validate(data)
InvalidDataError: The provided data does not match the metadata

Error: Invalid values found for numerical column 'age': ('a', 'b', 'c', +more)
Error: Key column 'user_id' contains missing values
Error: Foreign key column 'purchaser_id' contains unknown references: ('Unknown', 'USER_999', 'ZZZ', +more). All the values in must reference a primary key.
```

Reply:
The top line is for the overall error we throw (afaik, Python only allows you to throw 1 error at a time). The rest of the rows are the message.

Breaking up by table makes sense for cleanly summarizing where the errors happen. This could be similar to what we do in `metadata.validate()`: We split it up by table and then add a section for relationships.

```python
>>> synthesizer.validate(data)
InvalidDataError: The provided data does not match the metadata

Table: 'sessions'
Error: Invalid values found for numerical column 'age': ('a', 'b', 'c', +more)

Table: 'users'
Error: Key column 'user_id' contains missing values
...

Relationships:
Error: Foreign key column 'purchaser_id' contains unknown references: ('Unknown', 'USER_999', 'ZZZ', +more). All the values in must reference a primary key.
```

For context, I'm pasting below what we do for `metadata.validate()`:
```python
InvalidMetadataError: The metadata is not valid

Table: 'users'
Error: Invalid values ("pii") for datetime column "start_date".
Error: A Unique constraint is being applied to column "age". This column is already a key for that table.

Table: 'transactions'
Error: Invalid regex format string "[A-{6}" for text column "transaction_id"
Error: Unknown key value 'ttid'. Keys should be columns that exist in the table.
Error: Invalid increment value (0.5) in a FixedIncrements constraint. Increments must be positive integers.

Relationships:
Error: Relationship between tables ('users', 'transactions') contains an unknown primary key 'userr_id'.
Error: The relationships in the dataset are disjointed. Tables ('sessions') are not connected to any of the other tables.

```